src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java

```diff
@@ -0,0 +1,580 @@
+                    DateUtils.addDaysToDate(
+                            new Date(),
+                            mCapabilities.getFilesSharingPublicExpireDateDays()
+                    )
+            );
+
+            getExpirationDateValue(view).setVisibility(View.VISIBLE);
+
+            getExpirationDateValue(view).setText(formattedDate);
+        }
+
+        // Fill in the different fields if the share is being updated
+        if (updating()) {
+
+            // Set dialog title to edit
+            getDialogTitle(view).setText(R.string.share_via_link_edit_title);
+
+            // Set existing share name
+            getNameValue(view).setText(mPublicShare.getName());
+
+            // Set the edit permissions, if any
+            if (mPublicShare.getPermissions() == CRUD_PERMISSIONS) {
+                getEditPermissionSwitch(view).setChecked(true);
+            }
+
+            // Set password, if any
+            if (mPublicShare.isPasswordProtected()) {
+
+                // Switch on the password toggle
+                getPasswordSwitch(view).setChecked(true);
+
+                getPasswordValue(view).setVisibility(View.VISIBLE);
+
+                // Set an example password
+                getPasswordValue(view).setHint(R.string.share_via_link_default_password);
+            }
+
+            //Set expiration date, if any
+            if (mPublicShare.getExpirationDate() != 0) {
+
+                // Switch on the expiration date toggle
+                getExpirationDateSwitch(view).setChecked(true);
+
+                String formattedDate =
+                        SimpleDateFormat.getDateInstance().format(
+                                new Date(mPublicShare.getExpirationDate())
+                        );
+
+                getExpirationDateValue(view).setVisibility(View.VISIBLE);
+
+                // Set the existing share expiration date
+                getExpirationDateValue(view).setText(formattedDate);
+            }
+        }
+
+        // Set listener for user actions on password
+        initPasswordListener(view);
+
+        // Set listener for user actions on expiration date
+        initExpirationListener(view);
+
+        // Confirm add public link
+        Button confirmAddPublicLinkButton = (Button) view.findViewById(R.id.confirmAddPublicLinkButton);
+
+        confirmAddPublicLinkButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+
+                String publicLinkName = getNameValue(getView()).getText().toString();
+
+                String publicLinkPassword = getPasswordValue(getView()).getText().toString();
+
+                String expirationDate = getExpirationDateValue(getView()).getText().toString();
+
+                long publicLinkExpirationDateMillis = -1;
+
+                // Parse expiration date and convert it to milliseconds
+                if (expirationDate != null) {
```

Comment:
This is always 'true' (see the warning in Android Studio). The condition can be deleted safely.